### ISSUE TITLE
Order not rejected/accepted

### ISSUE THREAD

**Issue by bonomat** [itchysats/itchysats#1302]
Version: `maker:latest` and taker: 10ae1fb98c152e1d323edd22913d6b5378368287

- Tried to take an offer.
- First attempt crashed the VM
- After restart order is still in DB and automation tries to reject it
- rejection is not recorded/saved and maker keeps trying to reject it

```
  | 2022-02-02 09:48:26 | {"timestamp":"2022-02-01T22:48:26.264524964Z","level":"DEBUG","fields":{"message":"Maker rejects order","order_id":"842c1def-9e52-4b97-a24c-82581100ef0c"},"target":"daemon::maker_cfd"}
```

**Comment by bonomat** [comment]
Maybe this is related: 

```json

   {   "timestamp": "2022-02-01T22:39:56.094100505Z",   "level": "WARN",   "fields": {     "message": "select name, data, created_at …; rows: 71, elapsed: 1.065s\n\nselect\n  name,\n  data,\n  created_at as \"created_at: crate::model::Timestamp\"\nfrom\n  events\nwhere\n  cfd_id = $1\n",     "log.target": "sqlx::query",     "log.module_path": "sqlx::query"   },   "target": "sqlx::query" }
```

**Comment by bonomat** [comment]
Analysis notes: 

We see the error message 
`No active contract setup for order ...` as printed here: 

https://github.com/itchysats/itchysats/blob/249e2439ed0a3d5d78d578ab0e7721083a394186/daemon/src/maker_cfd.rs#L339-L342

We don't see
`Failed to reject order to taker` from here: 

https://github.com/itchysats/itchysats/blob/249e2439ed0a3d5d78d578ab0e7721083a394186/daemon/src/setup_maker.rs#L188-L195

Assumption: 
we fail at sending `RejectOrder` to taker and hence never update our local state. 
Not sure why we don't see the error message though...

**Comment by bonomat** [comment]
Proposal: 

Rejecting an order should not fail on the maker side. We should always mark the order as rejected, otherwise we might end up in an endless loop.  
I would just log the message but try to update the state in the db. 

This might not be optimal for the taker as an order might never gets rejected/accepted. Do we have a timeout here?

Thoughts?

**Comment by da-kami** [comment]
I can conform the behaviour, I run into the same issue with old, failed orders:

```
2022-02-02 10:13:29 DEBUG Sending action url=http://localhost:8001/api/cfd/17b6abc4-3e5a-4157-8520-f6f81cb81824/rejectOrder body="rejectOrder"
2022-02-02 10:13:29 ERROR Could not post action: {"status":500,"title":"RejectOrder failed","detail":"No active contract setup for order 17b6abc4-3e5a-4157-8520-f6f81cb81824: setup_maker::Actor actor is down"}
2022-02-02 10:13:29  INFO Rejecting order request due to "Requested price is below threshold. 37224.56 <= 38607.8535" order_id=679c18c7-2363-4031-a318-038e7814121f
2022-02-02 10:13:29 DEBUG Sending action url=http://localhost:8001/api/cfd/679c18c7-2363-4031-a318-038e7814121f/rejectOrder body="rejectOrder"
```

**Comment by da-kami** [comment]
> Assumption:
> we fail at sending RejectOrder to taker and hence never update our local state.
> Not sure why we don't see the error message though...

We fail earlier. I think we never enter the code in `itchysats/daemon/src/setup_maker.rs` because that actor does not exist (anymore).

**Comment by da-kami** [comment]
> Rejecting an order should not fail on the maker side. We should always mark the order as rejected, otherwise we might end up in an endless loop.
> I would just log the message but try to update the state in the db.

We would have to deal with errors on different levels. At the moment the setup actor is expected to be available when
I 

> This might not be optimal for the taker as an order might never gets rejected/accepted. Do we have a timeout here?

Not at the moment, After a quick scan I think we would have to add a timeout here to be safe:

https://github.com/itchysats/itchysats/blob/249e2439ed0a3d5d78d578ab0e7721083a394186/daemon/src/setup_taker.rs#L173-L193


> I can conform the behaviour, I run into the same issue with old, failed orders:
> 
> ```
> 2022-02-02 10:13:29 DEBUG Sending action url=http://localhost:8001/api/cfd/17b6abc4-3e5a-4157-8520-f6f81cb81824/rejectOrder body="rejectOrder"
> 2022-02-02 10:13:29 ERROR Could not post action: {"status":500,"title":"RejectOrder failed","detail":"No active contract setup for order 17b6abc4-3e5a-4157-8520-f6f81cb81824: setup_maker::Actor actor is down"}
> 2022-02-02 10:13:29  INFO Rejecting order request due to "Requested price is below threshold. 37224.56 <= 38607.8535" order_id=679c18c7-2363-4031-a318-038e7814121f
> 2022-02-02 10:13:29 DEBUG Sending action url=http://localhost:8001/api/cfd/679c18c7-2363-4031-a318-038e7814121f/rejectOrder body="rejectOrder"
> ```

This is actually a different scenario than the one described by @bonomat

**Comment by thomaseizinger** [comment]
> Rejecting an order should not fail on the maker side. We should always mark the order as rejected, otherwise we might end up in an endless loop.

Agree that updating local state should take priority.

**Comment by thomaseizinger** [comment]
This has been fixed right?

**Comment by bonomat** [comment]
Should have been fixed by @da-kami and me in #1307
